src/main/docs/guide/httpServer/serverConfiguration/threadPools/atBlocking.adoc

```diff
@@ -0,0 +1,21 @@
+You can use the ann:core.annotation.Blocking[] annotation to mark methods as blocking.
+
+If you set `micronaut.server.thread-selection` to `AUTO`, The Micronaut Framework offloads the execution of methods annotated with `@Blocking` to the IO thread pool (See: api:io.micronaut.scheduling.TaskExecutors[]).
+
+NOTE: `@Blocking` only works if you are using `AUTO` thread selection. Micronaut Framework defaults to `MANUAL` thread selection since Micronaut 2.0. We recommend the usage of ann:scheduling.annotation.ExecuteOn[] annotation to execute the blocking operations on a different thread. `@ExecutesOn` works for both `MANUAL` or `AUTO` thread selection.
+
+There are some places where the Micronaut framework uses ann:core.annotation.Blocking[] already.
```

Comment:
```suggestion
There are some places where the Micronaut framework uses ann:core.annotation.Blocking[] internally:
```

Reply:
why do you want to add internally? It is public facing API.

Reply:
What I mean is that it's implementation details. This piece of documentation is for educating users about how certain things are _internally_ implemented. It is a public API, yes, but still it's implementation details that won't affect them nor something they need to handle specifically.